src/msw/window.cpp

```diff
@@ -6276,6 +6331,40 @@ bool wxWindowMSW::HandlePressAndTap(const wxPoint& pt, WXDWORD flags)
 
     return HandleWindowEvent(event);
 }
+
+bool wxWindowMSW::HandleTouch(WXWPARAM wParam, WXLPARAM lParam)
+{
+    const unsigned count = LOWORD(wParam);
+    wxVector<TOUCHINPUT> info(count);
+    if (GestureFuncs::GetTouchInputInfo() ((HTOUCHINPUT)lParam, count, &info[0], sizeof(TOUCHINPUT)))
+    {
+        for(unsigned i = 0; i < count; i++)
+        {
+            // hundredths of a pixel of physical screen coordinates
+            wxPoint pos = ScreenToClient(wxPoint(info[i].x / 100, info[i].y / 100));
+            wxEventType type;
+            if ( info[i].dwFlags & TOUCHEVENTF_DOWN )
+                type = wxEVT_TOUCH_BEGIN;
+            else if ( info[i].dwFlags & TOUCHEVENTF_MOVE )
+                type = wxEVT_TOUCH_MOVE;
+            else if ( info[i].dwFlags & TOUCHEVENTF_UP )
+                type = wxEVT_TOUCH_END;
+            else
+                continue;
+
+            wxMultiTouchEvent event( GetId(), type);
+
+            event.SetEventObject( this );
+            event.SetPosition(pos);
+            event.SetSequenceId(wxTouchSequenceId(wxUIntToPtr(info[i].dwID + 1)));
+            event.SetPrimary( (info[i].dwFlags & TOUCHEVENTF_PRIMARY) != 0 );
+            HandleWindowEvent(event);
```

Comment:
Shouldn't we remember if any of the events were processed and only return true in this case?

Reply:
This also seems to be still relevant.